File: csharp/SlimDownYourAggregates.Tests/Slimmed/Entities/Author.cs
namespace SlimDownYourAggregates.Tests.Slimmed.Entities;

public class Author
{
    public Author(string firstName, string lastName)
    {
        FirstName = firstName;
        LastName = lastName;
    }

    public string FirstName { get; }
    public string LastName { get; }
}

File: java/src/main/java/io/eventdriven/slimdownaggregates/original/domain/books/entities/Author.java
package io.eventdriven.slimdownaggregates.original.domain.books.entities;

import static io.eventdriven.slimdownaggregates.original.infrastructure.validation.Validation.assertNotNull;

public record Author(
  AuthorId id,
  AuthorFirstName firstName,
  AuthorLastName lastName
) {
  public Author{
    assertNotNull(id);
    assertNotNull(firstName);
    assertNotNull(lastName);
  }
}
